src/nvim/fileio.c

```diff
@@ -1423,52 +1423,32 @@ readfile (
               }
             }
           }
-          if (enc_utf8) {               /* produce UTF-8 */
-            dest -= utf_char2len(u8c);
-            (void)utf_char2bytes(u8c, dest);
-          } else {                    /* produce Latin1 */
-            --dest;
-            if (u8c >= 0x100) {
-              /* character doesn't fit in latin1, retry with
-               * another fenc when possible, otherwise just
-               * report the error. */
-              if (can_retry)
-                goto rewind_retry;
-              if (conv_error == 0)
-                conv_error = readfile_linenr(linecnt, ptr, p);
-              if (bad_char_behavior == BAD_DROP)
-                ++dest;
-              else if (bad_char_behavior == BAD_KEEP)
-                *dest = u8c;
-              else if (eap != NULL && eap->bad_char != 0)
-                *dest = bad_char_behavior;
-              else
-                *dest = 0xBF;
-            } else
-              *dest = u8c;
-          }
+          assert(u8c <= INT_MAX);
+          dest -= utf_char2len((int)u8c);
```

Comment:
The comment should be kept. Even though it's somewhat redundant since the branch was removed, it still seems useful to hint that this is where "produce" happens.

```suggestion
          // produce UTF-8
          dest -= utf_char2len((int)u8c);
```